practice/tic-tac-toe/dict_board.py

```diff
@@ -0,0 +1,100 @@
+"""This module performs the functions for Tic-Tac-Toe (TTT) program of placing a token on the board, determining the
+winner of a game, and returning a str version of the board, using dict data type.
+"""
+
+WINNING_COMBINATIONS = [[(0, 0), (1, 0), (2, 0)], [(0, 0), (0, 1), (0, 2)], [(1, 0), (1, 1), (1, 2)], [(0, 1), (1, 1),
+                                                                                                       (1, 2)],
+                        [(2, 0), (2, 1), (2, 2)], [(0, 2), (1, 2), (2, 2)], [(0, 0), (1, 1), (2, 2)], [(2, 0), (1, 1),
+                                                                                                       (0, 2)]]
+
+class DictTTTBoard:
+    """Contains a blank TTT board as dict items, and commands to modify the board, score game, and return str version
+    of board.
+    """
+    def __init__(self):
+        """Defines input value."""
+        self._tokens_to_coords = {}
+
+    def __repr__(self):
+        """Returns real version.
+
+        >>> repr(DictTTTBoard())
+        'DictTTTBoard()'
+        """
+        return 'DictTTTBoard()'
+
+    def __eq__(self, other):
+        """Defines eauality.
+
+        >>> DictTTTBoard() == DictTTTBoard()
+        True
+        """
+        return self._tokens_to_coords == other._tokens_to_coords
+
+    def place_token(self, x, y, token):
+        """Uses grouping to add token as a new key item in a dictionary, all coordinates associating with that token are
+        stored in the value as a list of tuples.
+
+        >>> X = DictTTTBoard()
+        >>> X.place_token(0, 0, 'X')
+        >>> X._tokens_to_coords
+        {'X': [(0, 0)]}
+        """
+        coords = (x, y)
+        key = token
+        self._append_to_tokens_to_coords(key, coords)
+
+    def calc_winner(self):
+        """Calculates what token character string has won or None if no one has.
+
+        >>> X = DictTTTBoard()
+        >>> X._tokens_to_coords = {'X': [(0, 0), (1, 0), (2, 0)]}
+        >>> X.calc_winner()
+        'X'
+        >>> X = DictTTTBoard()
+        >>> X._tokens_to_coords = {'X': [(0, 0), (1, 0)]}
+        >>> X.calc_winner() is None
+        True
+        """
+        winner = None
+        for item in self._tokens_to_coords:
+            if sorted(self._tokens_to_coords[item]) in WINNING_COMBINATIONS:
+                winner = item
+        return winner
```

Comment:
In a lot of your code you use `item` vs `token` and having uniform naming helps a lot to understand this code.